pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,93 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+type SiteInfo struct {
+	// Edge cluster path on which the networking elements will be created.
+	EdgeClusterPaths []string `json:"edgeClusterPaths,omitempty"`
+}
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration.
+// There is a default VPCNetworkConfiguration that applies to Namespaces
+// do not have a VPCNetworkConfiguration assigned. When a field is not set
+// in a Namespace's VPCNetworkConfiguration, the Namespace will use the value
+// in the default VPCNetworkConfiguration.
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied.
+	// The default VPCNetworkConfiguration will not have AppliedToNamespaces set.
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// PolicyPath of Tier0 or Tier0 VRF gateway.
+	DefaultGatewayPath string `json:"defaultGatewayPath,omitempty"`
+	// Collection of Site information.
+	// +kubebuilder:validation:MaxItems=1
+	SiteInfos []SiteInfo `json:"siteInfos,omitempty"`
```

Comment:
Why we need this? The whole Supervisor belongs to a single site for sure.

Reply:
Yes, we only support single site on Prem.
The NSX VPC API use this, so just want to keep consistent with NSX VPC API structure. Do you think we can just expose one edge_cluster_path? @jianjuns 

 {
        "display_name": "vpc-2",
        "site_infos": [
             {
                "edge_cluster_paths": [
                    "/infra/sites/default/enforcement-points/default/edge-clusters/4a033a2f-d42e-421d-a41b-33333656278a"
                ],
                "site_path": "/infra/sites/default"
            }
        ],

Reply:
This CRD is from vCenter API. We need not be exactly the same as VPC configuration.

Yes let us remove this field.

Reply:
thanks Jianjun, removed site_info. Only 1 edgeClusterPath.